ifrs17/Utils/Queries.ipynb

```diff
@@ -399,6 +400,74 @@
             "metadata": {},
             "execution_count": 0,
             "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Raw and Ifrs Variables"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> LoadVariablesAsync<T>(this IDataSource querySource, Guid partition) where T : BaseDataRecord",
+                "\n{",
+                "\n    await querySource.Partition.SetAsync<PartitionByReportingNodeAndPeriod>(partition);",
+                "\n    return (await querySource.Query<T>().ToArrayAsync());",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public static async Task<T[]> LoadVariablesRelaxedAsync<T>(this IWorkspace workspace,",
+                "\n    IDataSource dataSource,",
+                "\n    Guid targetPartition,",
+                "\n    Guid defaultPartition)",
+                "\n    where T : BaseDataRecord",
+                "\n{",
+                "\n    var variables = await workspace.LoadVariablesAsync<T>(targetPartition);",
+                "\n    if (!variables.Any() && targetPartition != defaultPartition) {",
```

Comment:
This whole if cascade looks a bit fishy... you seem to be doing the 1st and 3rd statement again, but the variables might have changed in between